Microsoft.Toolkit.Uwp.UI.Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,312 @@
+                    Path = new PropertyPath(nameof(ItemHeaderTemplate)),
+                    Mode = BindingMode.OneWay
+                };
+                tvi.SetBinding(TabViewItem.HeaderTemplateProperty, headertemplatebinding);
+            }
+
+            if (tvi.ReadLocalValue(TabViewItem.IsClosableProperty) == DependencyProperty.UnsetValue)
+            {
+                var iscloseablebinding = new Binding()
+                {
+                    Source = this,
+                    Path = new PropertyPath(nameof(CanCloseTabs)),
+                    Mode = BindingMode.OneWay,
+                };
+                tvi.SetBinding(TabViewItem.IsClosableProperty, iscloseablebinding);
+            }
+
+            base.PrepareContainerForItemOverride(element, item);
+        }
+
+        private void TabViewItem_Loaded(object sender, RoutedEventArgs e)
+        {
+            var tvi = sender as TabViewItem;
+
+            tvi.Loaded -= TabViewItem_Loaded;
+
+            // Only need to do this once.
+            if (!hasLoaded)
+            {
+                hasLoaded = true;
+
+                // Need to set a tab's selection on load, otherwise ListView resets to null.
+                SetInitialSelection();
+
+                // Need to make sure ContentPresenter is set to content based on selection.
+                TabView_SelectionChanged(this, null);
+
+                // Need to make sure we've registered our removal method.
+                ItemsSource_PropertyChanged(this, null);
+
+                // Make sure we complete layout now.
+                TabView_SizeChanged(this, null);
+            }
+        }
+
+        private void TabViewItem_Closing(object sender, TabClosingEventArgs e)
+        {
+            var item = ItemFromContainer(e.Tab);
+
+            var args = new TabClosingEventArgs(item, e.Tab);
+            TabClosing?.Invoke(this, args);
+
+            if (!args.Cancel)
+            {
+                if (ItemsSource != null)
+                {
+                    _removeItemsSourceMethod?.Invoke(ItemsSource, new object[] { item });
+                }
+                else
+                {
+                    Items.Remove(item);
+                }
+            }
+        }
+
+        private void TabView_DragItemsStarting(object sender, DragItemsStartingEventArgs e)
+        {
+            // Keep track of drag so we don't modify content until done.
+            isDragging = true;
+        }
+
+        private void TabView_DragItemsCompleted(ListViewBase sender, DragItemsCompletedEventArgs args)
+        {
+            isDragging = false;
+
+            // args.DropResult == None when outside of area (e.g. create new window)
+            if (args.DropResult == DataPackageOperation.None)
+            {
+                var item = args.Items.FirstOrDefault();
+                TabDraggedOutside?.Invoke(this, new TabDraggedOutsideEventArgs(item, ContainerFromItem(item) as TabViewItem));
```

Comment:
`ContainerFromItem` is returning null when item is dragged outside of the TabView. This is causing the sample app to crash because the `Tab` property is null

Reply:
Good catch! @michael-hawker

Reply:
oops, I changed that at the last bit, I'll double-check and fix.  thanks